### ISSUE TITLE
gradle/gradle should work with native IDEA import and module per source set option

### ISSUE THREAD

**Issue by wolfs** [gradle/gradle#5799]
This way we would break the cycles between the modules and we would be able to set a different source compatibility per module.

This depends on #3980.

**Comment by wolfs** [comment]
We probably need to add a few directories when we create the classpath when running from IDEA: https://github.com/gradle/gradle/blob/188585eb1dc95f70d35d380c9e0ff5f6396fbf67/subprojects/core/src/main/java/org/gradle/api/internal/classpath/DefaultModuleRegistry.java#L222-L243

**Comment by jjohannes** [comment]
I investigated this. It looks like, there is only little to adjust in our build to makes this work. It will also solve the issue that the classpath is often not correct in IDEA at the moment, because `main` and `testFixtures` are merged into one module and end up with the same classpath.

However, I hit two issues on the IDEA side:
- https://youtrack.jetbrains.com/issue/IDEA-202271
- Bug with additional output folder handling (TODO link youtrack)

We should wait until these issues are resolved before we continue here.

**Comment by jjohannes** [comment]
WIP branch: https://github.com/gradle/gradle/tree/jjohannes/dev-prod/idea-sync

**Comment by wolfs** [comment]
Can we merge the WIP branch? It only seems to clean up the generated resources configuration in `baseServices` and add the post sync tasks. I think it would be great to have the post-sync tasks already.

Can we make the build work when we use `jar` dependencies for test fixtures even while the two Intellij issues are not fixed?

**Comment by jjohannes** [comment]
Both IDEA issues have been fixed and the fixes will be in the next EAP release.

PR for the improvement/fixes in our build: #7768
